packages/components/src/components/Radio/Radio.tsx

```diff
@@ -0,0 +1,108 @@
+import React from "react";
+import { useUID } from "react-uid";
+import {
+  Item as RadioItem,
+  RadioGroupItemProps,
+  Indicator as RadioIndicator,
+} from "@radix-ui/react-radio-group";
+import { Box } from "../../primitives/Box";
+import { Text } from "../../primitives/Text";
+
+export interface RadioProps extends RadioGroupItemProps {
+  /** The label of the radio. */
+  label: string;
+  /** The value of the radio. */
+  value: string;
+  /** Sets the state of the radio to disabled. */
+  disabled?: boolean;
+  /** Sets the state of the radio to disabled. */
+  hasError?: boolean;
+}
+
+/** A Radio is one button of a set of checkable buttons—known as radio buttons—where no more than one of the buttons can be checked at a time. */
+const Radio = React.forwardRef<HTMLButtonElement, RadioProps>(
+  ({ disabled = false, hasError = false, label, value, ...props }, ref) => {
+    const radioID = useUID();
+    return (
+      <Box.div alignItems="center" display="flex" gap="space30">
+        <Box.button
+          appearance="none"
+          aria-invalid={hasError}
+          as={RadioItem}
+          backgroundColor={{
+            _: "colorBackground",
+            hover: "colorBackgroundWeak",
+            checked: {
+              _: hasError
+                ? "colorBackgroundDestructiveStrong"
+                : "colorBackgroundPrimary",
+              hover: hasError
+                ? "colorBackgroundDestructiveStrong"
+                : "colorBackgroundPrimaryStrong",
+              disabled: {
+                _: "colorBackgroundStrong",
+                hover: "colorBackgroundStrong",
+              },
+            },
+            disabled: {
+              _: "colorBackground",
+              hover: "colorBackground",
+            },
+          }}
+          borderColor={{
+            _: hasError ? "colorBorderError" : "colorBorder",
+            checked: hasError ? "colorBorderError" : "colorBorderPrimary",
+            disabled: "colorBorderWeaker",
+          }}
+          borderRadius="borderRadiusCircle"
+          borderStyle="borderStyleSolid"
+          borderWidth="borderWidth10"
+          disabled={disabled}
+          h="1rem"
+          id={radioID}
+          margin="space0"
```

Comment:
@blumaa @apvale We need this margin reset here. Safari likes to add margins to some buttons. Since the Radix primitive renders as a button, we'll need to remove any browser default styles.